### PR TITLE
Add notification channel to schedules

### PR DESCRIPTION
**Pull request by faiyaz26**

### CODE REVIEW

File: protos/schedules.proto [hunk 1]
```diff
@@ -2,12 +2,36 @@ syntax="proto3";
 
 package dataform.schedules;
 
+
+message EmailNotificationChannel {
+  repeated string emails = 1;
```
Comment (BenBirt) [inline protos/schedules.proto]: i think we should call this field `to`

File: protos/schedules.proto [hunk 2]
```diff
@@ -2,12 +2,36 @@ syntax="proto3";
...
+message NotificationChannel {
+ string name = 1;
+ oneof channel {
+    EmailNotificationChannel email_notification = 1;
+    SlackNotificationChannel slack_notification = 2;
+  } 	
+}
+
+message ScheduleNotification {
+ repeated environments = 1;
```
Comment (BenBirt) [inline protos/schedules.proto]: presumably this should be a `string`?

File: protos/schedules.proto [hunk 3]
```diff
@@ -2,12 +2,36 @@ syntax="proto3";
...
+ repeated string onEvent = 2;
```
Comment (BenBirt) [inline protos/schedules.proto]: protobuf fields are snake_case

Comment (BenBirt) [inline protos/schedules.proto]: we should put a comment on this, pointing at the enum of possible values

File: protos/schedules.proto [hunk 4]
```diff
@@ -20,11 +44,15 @@ message Schedule {
   bool disabled = 2;
   ScheduleOptions options = 3;
   string cron = 4; // required field
+  // to support legacy
```
Comment (BenBirt) [inline protos/schedules.proto]: `// Deprecated`

File: protos/schedules.proto [hunk 5]
```diff
@@ -2,12 +2,36 @@ syntax="proto3";
...
+
+message NotificationChannel {
+ string name = 1;
+ oneof channel {
+    EmailNotificationChannel email_notification = 1;
+    SlackNotificationChannel slack_notification = 2;
+  } 	
+}
+
+message ScheduleNotification {
```
Comment (BenBirt) [inline protos/schedules.proto]: @lewish now's the right time to make the final decision about whether we have a `schedules` field in here or not.

Comment (lewish) [reply protos/schedules.proto]: Personally, I'm happy with this...

File: protos/schedules.proto (hunk 3 above)
Comment (faiyaz26) [reply protos/schedules.proto]: Added an Enum, and added a comment here

File: protos/schedules.proto (hunk 5 above)
Comment (BenBirt) [reply protos/schedules.proto]: ok!

File: protos/schedules.proto [hunk 6]
```diff
@@ -2,12 +2,42 @@ syntax="proto3";
 
 package dataform.schedules;
 
+enum NotificationEventType {
+  ON_ANY = 0;
```
Comment (BenBirt) [inline protos/schedules.proto]: i don't think we need this value (it corresponds to a non-existent `events` list IMO)

File: protos/schedules.proto [hunk 7]
```diff
@@ -2,12 +2,42 @@ syntax="proto3";
...
+  ON_SUCCESS = 1;
```
Comment (BenBirt) [inline protos/schedules.proto]: IMO we should drop the "ON_" prefixes

File: protos/schedules.proto (hunk 6 above)
Comment (faiyaz26) [reply protos/schedules.proto]: I get error, that Enum should have a index of 0, what should I do about that?

Comment (lewish) [reply protos/schedules.proto]: Do we need this enum at all?
If so, I'd just call the first value - UKNOWN (that's what we do everywhere else).

File: protos/schedules.proto (hunk 7 above)
Comment (lewish) [reply protos/schedules.proto]: +1

**Pull request merged**